Game.java

```diff
@@ -1,19 +1,37 @@
+import java.util.ArrayList;
 import java.util.List;
 
 public class Game {
 
+  // names of players, set by constructor
   private String[] players = new String[4];
+  // used to terminate the while loop once a player reaches at least 100 points
   private boolean gameOver = false;
 
+  // position i has the cumulative scores of each player up to the end of ith
+  // round
```

Comment:
Consider making the `RoundResult` object, which is just a holder for 4 integers. This indirection makes the type easier to read (`List<RoundResult>`), and should include a 4-argument constructor that essentially guarantees that it will always be filled with exactly four scores, one from each player. It also abstracts away the round into its own black box, which increases isolation.

In fact, you can even have `CumulativeHistory` be its own object, offering various services like `int getNumberOfRounds()` or `RoundResult getCumulativeScores(int roundNumber)`. The method `addToGameScoresHistory` below would actually be better located as part of `RoundHistory` as a `void addRound(RoundResult roundResult)`. Now `Game` is an isolated concept from the management of cumulative scores, which is nice.

Reply:
I don't have a good sense of when to make a new class, as opposed to just an instance variable in another class.